presto-main/src/main/java/com/facebook/presto/operator/scalar/UrlFunctions.java

```diff
@@ -129,6 +132,42 @@ public static Slice urlExtractParameter(@SqlType(StandardTypes.VARCHAR) Slice ur
         return null;
     }
 
+    @Nullable
+    @Description("encode url")
+    @ScalarFunction
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice urlEncode(@SqlType(StandardTypes.VARCHAR) Slice url)
+    {
+        if (url == null) {
+            return null;
+        }
+
+        try {
+            return slice(URLEncoder.encode(url.toString(UTF_8), UTF_8.name()));
+        }
+        catch (UnsupportedEncodingException e) {
+            return null;
+        }
+    }
+
+    @Nullable
+    @Description("decode url")
+    @ScalarFunction
+    @SqlType(StandardTypes.VARCHAR)
+    public static Slice urlDecode(@SqlType(StandardTypes.VARCHAR) Slice url)
+    {
+        if (url == null) {
+            return null;
+        }
+
+        try {
+            return slice(URLDecoder.decode(url.toString(UTF_8), UTF_8.name()));
+        }
+        catch (UnsupportedEncodingException e) {
+            return null;
```

Comment:
Why not write our own? Percent encoding is a trivial algorithm and can work directly on a Slice.